torch/_torch_docs.py

```diff
@@ -5885,3 +5885,58 @@ def parse_kwargs(desc):
             [4, 5, 6],
             [4, 5, 6]])
 """)
+
+
+add_docstr(torch.cartesian_prod,
+           r"""
+cartesian_prod(tensors) -> seq
+
+Do cartesian product of the given sequence of tensors. The behavior is similar to
+python's `itertools.prod`. The difference is, if the arguments is a sequence of
+size :math:`k`, `itertools.prod` generate :math:`k`-tuples, while `torch.cartesian_prod`
+create :math:`k` output tensors.
+
+Arguments:
+    tensors (sequence of Tensors): sequence of scalars or 1 dimensional tensors.
+        Scalars will be treated as tensors of size :math:`(1,)` automatically.
+
+Returns:
+    seq (sequence of Tensors): If the input has :math:`k` tensors of size
+        :math:`(N_1,), (N_2,), \ldots , (N_k,)`, then the output would also has :math:`k` tensors,
+        where all tensors are of size :math:`N_1 \times N_2 \times \ldots \times N_k`.
+
+Example::
+
+    >>> a = torch.tensor([1, 2, 3])
+    >>> b = torch.tensor([4, 5])
+    >>> torch.cartesian_prod([a, b])
+    (tensor([1, 1, 2, 2, 3, 3]), tensor([4, 5, 4, 5, 4, 5]))
```

Comment:
This isn't analogous to itertools.product's behavior:
```
import itertools
list(itertools.product([1, 2, 3], [4, 5]))

[(1, 4), (1, 5), (2, 4), (2, 5), (3, 4), (3, 5)]
```
am I missing something?

Reply:
`(tensor([1, 1, 2, 2, 3, 3]), tensor([4, 5, 4, 5, 4, 5]))` is the unbinded version of `tensor([[1, 4], [1, 5], [2, 4], [2, 5], [3, 4], [3, 5]])` either version looks good to me. But I just feel that the unbinded version might be more convenient since I think these indices would be used separately. But the stacked version seems to be more natural.